### PR TITLE
Draft 3 size engine function

### PR DESCRIPTION
**Pull request by cjllanwarne**
Demonstrates the working of openWDL PR https://github.com/openwdl/wdl/pull/169

- [x] Rebase onto develop after 3432

### CODE REVIEW

**Comment by cjllanwarne** [comment]
Red thumb required because I refactored the `size` function in `IoFunctionSet` in `cromwell.backend`.

Previously it was implementing a bunch of parameter parsing and implementing the WDL Draft 2 semantics. Now it tells you the size of a file and the language versions implement the semantics around that.

File: wom/src/main/scala/wom/expression/WomExpression.scala [hunk 1]
```diff
@@ -37,7 +37,7 @@ trait IoFunctionSet {
   def stderr(params: Seq[Try[WomValue]]): Try[WomSingleFile]
   def glob(pattern: String): Future[Seq[String]]
   def listAllFilesUnderDirectory(dirPath: String): Future[Seq[String]]
-  def size(params: Seq[Try[WomValue]]): Future[WomFloat]
+  def size(path: String): Future[Double]
```
Comment (kshakir) [inline wom/src/main/scala/wom/expression/WomExpression.scala]: Change this to `Future[Long]` as there are no fractional bytes.

File: supportedBackends/tes/src/test/scala/cromwell/backend/impl/tes/TesRuntimeAttributesSpec.scala [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package cromwell.backend.impl.tes
 
 import cromwell.backend.validation.ContinueOnReturnCodeSet
-import cromwell.backend.{BackendConfigurationDescriptor, MemorySize, RuntimeAttributeDefinition, TestConfig}
+import cromwell.backend.{BackendConfigurationDescriptor, RuntimeAttributeDefinition, TestConfig}
```
Comment (jbuabbud) [inline supportedBackends/tes/src/test/scala/cromwell/backend/impl/tes/TesRuntimeAttributesSpec.scala]: I think you need to import wom.format.MemorySize

File: supportedBackends/spark/src/test/scala/cromwell/backend/impl/spark/SparkRuntimeAttributesSpec.scala [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package cromwell.backend.impl.spark
 
 import wdl.draft2.model.Draft2ImportResolver
-import cromwell.backend.{BackendWorkflowDescriptor, MemorySize}
+import cromwell.backend.BackendWorkflowDescriptor
 import wom.RuntimeAttributesKeys._
 import cromwell.core.labels.Labels
 import cromwell.core.{WorkflowId, WorkflowOptions}
```
Comment (jbuabbud) [inline supportedBackends/spark/src/test/scala/cromwell/backend/impl/spark/SparkRuntimeAttributesSpec.scala]: import wom.format.MemorySize

File: supportedBackends/tes/src/test/scala/cromwell/backend/impl/tes/TesRuntimeAttributesSpec.scala (hunk 2 above)
Comment (cjllanwarne) [reply supportedBackends/tes/src/test/scala/cromwell/backend/impl/tes/TesRuntimeAttributesSpec.scala]: Oops! Added.

File: supportedBackends/spark/src/test/scala/cromwell/backend/impl/spark/SparkRuntimeAttributesSpec.scala (hunk 3 above)
Comment (cjllanwarne) [reply supportedBackends/spark/src/test/scala/cromwell/backend/impl/spark/SparkRuntimeAttributesSpec.scala]: Added

**Pull request merged**